// @azure/arm-keyvault/dist-esm/src/models/index.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
/** Known values of {@link DeletionRecoveryLevel} that the service accepts. */
export var KnownDeletionRecoveryLevel;
(function (KnownDeletionRecoveryLevel) {
    KnownDeletionRecoveryLevel["Purgeable"] = "Purgeable";
    KnownDeletionRecoveryLevel["RecoverablePurgeable"] = "Recoverable+Purgeable";
    KnownDeletionRecoveryLevel["Recoverable"] = "Recoverable";
    KnownDeletionRecoveryLevel["RecoverableProtectedSubscription"] = "Recoverable+ProtectedSubscription";
})(KnownDeletionRecoveryLevel || (KnownDeletionRecoveryLevel = {}));
/** Known values of {@link JsonWebKeyType} that the service accepts. */
export var KnownJsonWebKeyType;
(function (KnownJsonWebKeyType) {
    KnownJsonWebKeyType["EC"] = "EC";
    KnownJsonWebKeyType["ECHSM"] = "EC-HSM";
    KnownJsonWebKeyType["RSA"] = "RSA";
    KnownJsonWebKeyType["RSAHSM"] = "RSA-HSM";
})(KnownJsonWebKeyType || (KnownJsonWebKeyType = {}));
/** Known values of {@link JsonWebKeyOperation} that the service accepts. */
export var KnownJsonWebKeyOperation;
(function (KnownJsonWebKeyOperation) {
    KnownJsonWebKeyOperation["Encrypt"] = "encrypt";
    KnownJsonWebKeyOperation["Decrypt"] = "decrypt";
    KnownJsonWebKeyOperation["Sign"] = "sign";
    KnownJsonWebKeyOperation["Verify"] = "verify";
    KnownJsonWebKeyOperation["WrapKey"] = "wrapKey";
    KnownJsonWebKeyOperation["UnwrapKey"] = "unwrapKey";
    KnownJsonWebKeyOperation["Import"] = "import";
})(KnownJsonWebKeyOperation || (KnownJsonWebKeyOperation = {}));
/** Known values of {@link JsonWebKeyCurveName} that the service accepts. */
export var KnownJsonWebKeyCurveName;
(function (KnownJsonWebKeyCurveName) {
    KnownJsonWebKeyCurveName["P256"] = "P-256";
    KnownJsonWebKeyCurveName["P384"] = "P-384";
    KnownJsonWebKeyCurveName["P521"] = "P-521";
    KnownJsonWebKeyCurveName["P256K"] = "P-256K";
})(KnownJsonWebKeyCurveName || (KnownJsonWebKeyCurveName = {}));
/** Known values of {@link SkuFamily} that the service accepts. */
export var KnownSkuFamily;
(function (KnownSkuFamily) {
    KnownSkuFamily["A"] = "A";
})(KnownSkuFamily || (KnownSkuFamily = {}));
/** Known values of {@link KeyPermissions} that the service accepts. */
export var KnownKeyPermissions;
(function (KnownKeyPermissions) {
    KnownKeyPermissions["All"] = "all";
    KnownKeyPermissions["Encrypt"] = "encrypt";
    KnownKeyPermissions["Decrypt"] = "decrypt";
    KnownKeyPermissions["WrapKey"] = "wrapKey";
    KnownKeyPermissions["UnwrapKey"] = "unwrapKey";
    KnownKeyPermissions["Sign"] = "sign";
    KnownKeyPermissions["Verify"] = "verify";
    KnownKeyPermissions["Get"] = "get";
    KnownKeyPermissions["List"] = "list";
    KnownKeyPermissions["Create"] = "create";
    KnownKeyPermissions["Update"] = "update";
    KnownKeyPermissions["Import"] = "import";
    KnownKeyPermissions["Delete"] = "delete";
    KnownKeyPermissions["Backup"] = "backup";
    KnownKeyPermissions["Restore"] = "restore";
    KnownKeyPermissions["Recover"] = "recover";
    KnownKeyPermissions["Purge"] = "purge";
})(KnownKeyPermissions || (KnownKeyPermissions = {}));
/** Known values of {@link SecretPermissions} that the service accepts. */
export var KnownSecretPermissions;
(function (KnownSecretPermissions) {
    KnownSecretPermissions["All"] = "all";
    KnownSecretPermissions["Get"] = "get";
    KnownSecretPermissions["List"] = "list";
    KnownSecretPermissions["Set"] = "set";
    KnownSecretPermissions["Delete"] = "delete";
    KnownSecretPermissions["Backup"] = "backup";
    KnownSecretPermissions["Restore"] = "restore";
    KnownSecretPermissions["Recover"] = "recover";
    KnownSecretPermissions["Purge"] = "purge";
})(KnownSecretPermissions || (KnownSecretPermissions = {}));
/** Known values of {@link CertificatePermissions} that the service accepts. */
export var KnownCertificatePermissions;
(function (KnownCertificatePermissions) {
    KnownCertificatePermissions["All"] = "all";
    KnownCertificatePermissions["Get"] = "get";
    KnownCertificatePermissions["List"] = "list";
    KnownCertificatePermissions["Delete"] = "delete";
    KnownCertificatePermissions["Create"] = "create";
    KnownCertificatePermissions["Import"] = "import";
    KnownCertificatePermissions["Update"] = "update";
    KnownCertificatePermissions["Managecontacts"] = "managecontacts";
    KnownCertificatePermissions["Getissuers"] = "getissuers";
    KnownCertificatePermissions["Listissuers"] = "listissuers";
    KnownCertificatePermissions["Setissuers"] = "setissuers";
    KnownCertificatePermissions["Deleteissuers"] = "deleteissuers";
    KnownCertificatePermissions["Manageissuers"] = "manageissuers";
    KnownCertificatePermissions["Recover"] = "recover";
    KnownCertificatePermissions["Purge"] = "purge";
    KnownCertificatePermissions["Backup"] = "backup";
    KnownCertificatePermissions["Restore"] = "restore";
})(KnownCertificatePermissions || (KnownCertificatePermissions = {}));
/** Known values of {@link StoragePermissions} that the service accepts. */
export var KnownStoragePermissions;
(function (KnownStoragePermissions) {
    KnownStoragePermissions["All"] = "all";
    KnownStoragePermissions["Get"] = "get";
    KnownStoragePermissions["List"] = "list";
    KnownStoragePermissions["Delete"] = "delete";
    KnownStoragePermissions["Set"] = "set";
    KnownStoragePermissions["Update"] = "update";
    KnownStoragePermissions["Regeneratekey"] = "regeneratekey";
    KnownStoragePermissions["Recover"] = "recover";
    KnownStoragePermissions["Purge"] = "purge";
    KnownStoragePermissions["Backup"] = "backup";
    KnownStoragePermissions["Restore"] = "restore";
    KnownStoragePermissions["Setsas"] = "setsas";
    KnownStoragePermissions["Listsas"] = "listsas";
    KnownStoragePermissions["Getsas"] = "getsas";
    KnownStoragePermissions["Deletesas"] = "deletesas";
})(KnownStoragePermissions || (KnownStoragePermissions = {}));
/** Known values of {@link NetworkRuleBypassOptions} that the service accepts. */
export var KnownNetworkRuleBypassOptions;
(function (KnownNetworkRuleBypassOptions) {
    KnownNetworkRuleBypassOptions["AzureServices"] = "AzureServices";
    KnownNetworkRuleBypassOptions["None"] = "None";
})(KnownNetworkRuleBypassOptions || (KnownNetworkRuleBypassOptions = {}));
/** Known values of {@link NetworkRuleAction} that the service accepts. */
export var KnownNetworkRuleAction;
(function (KnownNetworkRuleAction) {
    KnownNetworkRuleAction["Allow"] = "Allow";
    KnownNetworkRuleAction["Deny"] = "Deny";
})(KnownNetworkRuleAction || (KnownNetworkRuleAction = {}));
/** Known values of {@link VaultProvisioningState} that the service accepts. */
export var KnownVaultProvisioningState;
(function (KnownVaultProvisioningState) {
    KnownVaultProvisioningState["Succeeded"] = "Succeeded";
    KnownVaultProvisioningState["RegisteringDns"] = "RegisteringDns";
})(KnownVaultProvisioningState || (KnownVaultProvisioningState = {}));
/** Known values of {@link PrivateEndpointServiceConnectionStatus} that the service accepts. */
export var KnownPrivateEndpointServiceConnectionStatus;
(function (KnownPrivateEndpointServiceConnectionStatus) {
    KnownPrivateEndpointServiceConnectionStatus["Pending"] = "Pending";
    KnownPrivateEndpointServiceConnectionStatus["Approved"] = "Approved";
    KnownPrivateEndpointServiceConnectionStatus["Rejected"] = "Rejected";
    KnownPrivateEndpointServiceConnectionStatus["Disconnected"] = "Disconnected";
})(KnownPrivateEndpointServiceConnectionStatus || (KnownPrivateEndpointServiceConnectionStatus = {}));
/** Known values of {@link ActionsRequired} that the service accepts. */
export var KnownActionsRequired;
(function (KnownActionsRequired) {
    KnownActionsRequired["None"] = "None";
})(KnownActionsRequired || (KnownActionsRequired = {}));
/** Known values of {@link PrivateEndpointConnectionProvisioningState} that the service accepts. */
export var KnownPrivateEndpointConnectionProvisioningState;
(function (KnownPrivateEndpointConnectionProvisioningState) {
    KnownPrivateEndpointConnectionProvisioningState["Succeeded"] = "Succeeded";
    KnownPrivateEndpointConnectionProvisioningState["Creating"] = "Creating";
    KnownPrivateEndpointConnectionProvisioningState["Updating"] = "Updating";
    KnownPrivateEndpointConnectionProvisioningState["Deleting"] = "Deleting";
    KnownPrivateEndpointConnectionProvisioningState["Failed"] = "Failed";
    KnownPrivateEndpointConnectionProvisioningState["Disconnected"] = "Disconnected";
})(KnownPrivateEndpointConnectionProvisioningState || (KnownPrivateEndpointConnectionProvisioningState = {}));
/** Known values of {@link IdentityType} that the service accepts. */
export var KnownIdentityType;
(function (KnownIdentityType) {
    KnownIdentityType["User"] = "User";
    KnownIdentityType["Application"] = "Application";
    KnownIdentityType["ManagedIdentity"] = "ManagedIdentity";
    KnownIdentityType["Key"] = "Key";
})(KnownIdentityType || (KnownIdentityType = {}));
/** Known values of {@link ProvisioningState} that the service accepts. */
export var KnownProvisioningState;
(function (KnownProvisioningState) {
    /** The managed HSM Pool has been full provisioned. */
    KnownProvisioningState["Succeeded"] = "Succeeded";
    /** The managed HSM Pool is currently being provisioned. */
    KnownProvisioningState["Provisioning"] = "Provisioning";
    /** Provisioning of the managed HSM Pool has failed. */
    KnownProvisioningState["Failed"] = "Failed";
    /** The managed HSM Pool is currently being updated. */
    KnownProvisioningState["Updating"] = "Updating";
    /** The managed HSM Pool is currently being deleted. */
    KnownProvisioningState["Deleting"] = "Deleting";
    /** The managed HSM pool is ready for normal use. */
    KnownProvisioningState["Activated"] = "Activated";
    /** The managed HSM pool is waiting for a security domain restore action. */
    KnownProvisioningState["SecurityDomainRestore"] = "SecurityDomainRestore";
    /** The managed HSM pool is being restored from full HSM backup. */
    KnownProvisioningState["Restoring"] = "Restoring";
})(KnownProvisioningState || (KnownProvisioningState = {}));
/** Known values of {@link PublicNetworkAccess} that the service accepts. */
export var KnownPublicNetworkAccess;
(function (KnownPublicNetworkAccess) {
    KnownPublicNetworkAccess["Enabled"] = "Enabled";
    KnownPublicNetworkAccess["Disabled"] = "Disabled";
})(KnownPublicNetworkAccess || (KnownPublicNetworkAccess = {}));
/** Known values of {@link ManagedHsmSkuFamily} that the service accepts. */
export var KnownManagedHsmSkuFamily;
(function (KnownManagedHsmSkuFamily) {
    KnownManagedHsmSkuFamily["B"] = "B";
})(KnownManagedHsmSkuFamily || (KnownManagedHsmSkuFamily = {}));
