// @azure/arm-keyvault/dist-esm/test/keyvault_examples.js
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
import { __asyncValues, __awaiter } from "tslib";
import { env, record, isPlaybackMode } from "@azure-tools/test-recorder";
import * as assert from "assert";
import { ClientSecretCredential } from "@azure/identity";
import { KeyVaultManagementClient } from "../src/keyVaultManagementClient";
const recorderEnvSetup = {
    replaceableVariables: {
        AZURE_CLIENT_ID: "azure_client_id",
        AZURE_CLIENT_SECRET: "azure_client_secret",
        AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
        SUBSCRIPTION_ID: "azure_subscription_id"
    },
    customizationsOnRecordings: [
        (recording) => recording.replace(/"access_token":"[^"]*"/g, `"access_token":"access_token"`)
    ],
    queryParametersToSkip: []
};
export const testPollingOptions = {
    updateIntervalInMs: isPlaybackMode() ? 0 : undefined,
};
describe("Keyvault test", () => {
    let recorder;
    let subscriptionId;
    let client;
    let location;
    let resourceGroup;
    let vaultName;
    let tenantId;
    beforeEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            recorder = record(this, recorderEnvSetup);
            subscriptionId = env.SUBSCRIPTION_ID;
            tenantId = env.AZURE_TENANT_ID;
            // This is an example of how the environment variables are used
            const credential = new ClientSecretCredential(env.AZURE_TENANT_ID, env.AZURE_CLIENT_ID, env.AZURE_CLIENT_SECRET);
            client = new KeyVaultManagementClient(credential, subscriptionId);
            location = "eastus";
            resourceGroup = "myjstest";
            vaultName = "myvaultzzzz";
        });
    });
    afterEach(function () {
        return __awaiter(this, void 0, void 0, function* () {
            yield recorder.stop();
        });
    });
    it("vaults create test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.beginCreateOrUpdateAndWait(resourceGroup, vaultName, {
                location: "eastus",
                properties: {
                    tenantId: tenantId,
                    sku: {
                        family: "A",
                        name: "standard",
                    },
                    accessPolicies: [
                        {
                            tenantId: tenantId,
                            objectId: "00000000-0000-0000-0000-000000000000",
                            permissions: {
                                keys: [
                                    "encrypt",
                                    "decrypt",
                                    "wrapKey",
                                    "unwrapKey",
                                    "sign",
                                    "verify",
                                    "get",
                                    "list",
                                    "create",
                                    "update",
                                    "import",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                secrets: [
                                    "get",
                                    "list",
                                    "set",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                certificates: [
                                    "get",
                                    "list",
                                    "delete",
                                    "create",
                                    "import",
                                    "update",
                                    "managecontacts",
                                    "getissuers",
                                    "listissuers",
                                    "setissuers",
                                    "deleteissuers",
                                    "manageissuers",
                                    "recover",
                                    "purge",
                                ],
                            },
                        },
                    ],
                    enabledForDeployment: true,
                    enabledForDiskEncryption: true,
                    enabledForTemplateDeployment: true,
                }
            }, testPollingOptions);
            assert.equal(res.name, vaultName);
        });
    });
    it("vaults get test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            if (isPlaybackMode()) {
                this.skip();
            }
            const res = yield client.vaults.get(resourceGroup, vaultName);
            assert.equal(res.name, vaultName);
        });
    });
    it("vaults list test", function () {
        var e_1, _a;
        return __awaiter(this, void 0, void 0, function* () {
            if (isPlaybackMode()) {
                this.skip();
            }
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.vaults.listByResourceGroup(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_1) throw e_1.error; }
            }
            assert.equal(resArray.length, 1);
        });
    });
    it("vaults update test", function () {
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.update(resourceGroup, vaultName, {
                properties: {
                    tenantId: tenantId,
                    sku: {
                        family: "A",
                        name: "standard",
                    },
                    accessPolicies: [
                        {
                            tenantId: tenantId,
                            objectId: "00000000-0000-0000-0000-000000000000",
                            permissions: {
                                keys: [
                                    "encrypt",
                                    "decrypt",
                                    "wrapKey",
                                    "unwrapKey",
                                    "sign",
                                    "verify",
                                    "get",
                                    "list",
                                    "create",
                                    "update",
                                    "import",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                secrets: [
                                    "get",
                                    "list",
                                    "set",
                                    "delete",
                                    "backup",
                                    "restore",
                                    "recover",
                                    "purge",
                                ],
                                certificates: [
                                    "get",
                                    "list",
                                    "delete",
                                    "create",
                                    "import",
                                    "update",
                                    "managecontacts",
                                    "getissuers",
                                    "listissuers",
                                    "setissuers",
                                    "deleteissuers",
                                    "manageissuers",
                                    "recover",
                                    "purge",
                                ],
                            },
                        },
                    ],
                    enabledForDeployment: true,
                    enabledForDiskEncryption: true,
                    enabledForTemplateDeployment: true,
                }
            });
            assert.equal(res.type, "Microsoft.KeyVault/vaults");
        });
    });
    it("vaults delete test", function () {
        var e_2, _a;
        return __awaiter(this, void 0, void 0, function* () {
            const res = yield client.vaults.delete(resourceGroup, vaultName);
            const resArray = new Array();
            try {
                for (var _b = __asyncValues(client.vaults.listByResourceGroup(resourceGroup)), _c; _c = yield _b.next(), !_c.done;) {
                    let item = _c.value;
                    resArray.push(item);
                }
            }
            catch (e_2_1) { e_2 = { error: e_2_1 }; }
            finally {
                try {
                    if (_c && !_c.done && (_a = _b.return)) yield _a.call(_b);
                }
                finally { if (e_2) throw e_2.error; }
            }
            assert.equal(resArray.length, 0);
        });
    });
});
